src/main/java/codeu/controller/DataParse.java

```
@@ -0,0 +1,82 @@
+    UUID convID = null;
+    Pattern p = Pattern.compile("(\\b[A-Z]{3,}\\b\\s?)+"); //Matches all-uppercase names
+
+    try (Scanner scanner = new Scanner(new File("/Users/stephanieyang/IdeaProjects/" +
+            "GoogleCodeU/src/main/java/codeu/controller/" + tf))) {
+      scanner.useDelimiter("\n");
+      while (scanner.hasNext()) {
+        String token = scanner.next();
+
+        if (token.contains("ACT")) {
+          continue;
+        } else if (token.contains("SCENE")) {
+          //Start of a new conversation
+          String nextToken = scanner.next();
+          Matcher m = p.matcher(nextToken);
+          if (m.find()) {
+            String userName = m.group(1);
+            UUID userID = findPerson(userName);
+            convID = UUID.randomUUID();
+            allConversations.add(new Conversation(convID, userID, token, Instant.now()));
+          }
+        } else if (token.matches("(\\b[A-Z]{3,}\\b\\s?)+") && !token.equals("PROLOGUE")) {
+          //Meaning a person is talking.
+          UUID userID = findPerson(token);
+          String text = "";
+          while (scanner.hasNext()) {
+            String nextLine = scanner.next();
+            if (nextLine.matches("(\\b[A-Z]{3,}\\b\\s?)+")) {
+              //Next line is a person responding.
+              allMessages.add(new Message(UUID.randomUUID(), convID, userID, text, Instant.now()));
+              text = "";
+              userID = findPerson(nextLine);
+            } else if (nextLine.isEmpty() || p.matcher(nextLine).find()) {
+              //Next line is either blank or an action, signifying the message's end.
+              allMessages.add(new Message(UUID.randomUUID(), convID, userID, text, Instant.now()));
+              break;
+            } else {
+              //Continue building up the message.
+              text += nextLine + "\n";
+            }
+          }
+        }
+      }
+    } catch (FileNotFoundException e) {
+        e.printStackTrace();
```

a useful pattern: 

throw new RuntimeException(e);

that way the compiler won't bother you with the checked exceptions but you still fail loudly if something is not working (which is usually better than printing error messages & proceeding since maybe no one will notice the errors being printed for a long time).